tools/topology/topology2/platform/intel/dmic1-passthrough.conf

```diff
@@ -54,13 +54,39 @@ Object.Pipeline.io-gateway-capture [
                         type		dai_out
                         stream_name	$DMIC1_NAME
                         node_type $DMIC_LINK_INPUT_CLASS
-                        num_input_audio_formats	2
+                        num_input_audio_formats	6
                         Object.Base.input_audio_format [
+                                {
+                                        in_rate			$DMIC1_RATE
+                                        in_bit_depth		16
+                                        in_valid_bit_depth	16
+                                }
+                                {
+                                        in_rate			$DMIC1_RATE
+                                        in_bit_depth		32
+                                        in_valid_bit_depth	24
+                                }
                                 {
                                         in_rate			$DMIC1_RATE
                                         in_bit_depth		32
                                         in_valid_bit_depth	32
                                 }
+                                {
+                                        in_rate			$DMIC1_RATE
+                                        in_channels		4
+                                        in_bit_depth		16
+                                        in_valid_bit_depth	16
+                                        in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+                                        in_ch_map	$CHANNEL_MAP_3_POINT_1
+                                }
+                                {
+                                        in_rate			$DMIC1_RATE
+                                        in_channels		4
+                                        in_bit_depth		32
+                                        in_valid_bit_depth	24
+                                        in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+                                        in_ch_map	$CHANNEL_MAP_3_POINT_1
+                                }
```

Comment:
After some back and forth, I'm leaning towards asking to add sample type explciitly here. I.e. "in_sample_type $SAMPLE_TYPE_MSB_INTEGER"

Reply:
which is only possible with minimal documentation... I wouldn't even know what to do with all this :-)

Reply:
Is this correct @kv2019i 

````
+-----------------------+-----------------------+------------------------------------+
|         SOF           |        ALSA           |       data representation          |
+-----------------------+-----------------------+------------------------------------+
|    SOF_S24_4LE_MSB    | S32_LE with msbits=24 |  0xA1234500 0xB1234500 0xC1234500  |
+-----------------------+-----------------------+------------------------------------+
|    SOF_S24_4LE_LSB	|  S24_LE               |  0x00A12345 0x00B12345 0x00C12345  |      
+-----------------------+-----------------------+------------------------------------+
|    SOF_S24_3LE        |  S24_3LE	        |  0xA12345B1 0x2345C123 0x45D12345  |
+-----------------------+-----------------------+------------------------------------+
|    SOF_S32_LE	        | S32_LE with msbits=32	|  0xA1234567 0xB1234567 0xC1234567  | 
+-----------------------+-----------------------+------------------------------------+
````

Reply:
Yep, this is good to add. Treating the data accidentally as LSB aligned 24 bits would be a disaster.